### PR TITLE
office-js: Add AsyncResult.diagnostics

### PR DESCRIPTION
**Pull request by ElizabethSamuel-MSFT**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/office-js/index.d.ts [hunk 1]
```diff
@@ -551,6 +551,37 @@ declare namespace Office {
         */
         asyncContext: any;
         /**
+        * Gets an object that may provide additional information if an error occurred.
```
Comment (AlexJerabek) [inline types/office-js/index.d.ts]: "Gets an object" -> "An object"
(Unless this is generated when called, but then this should be a method and not a property)

Comment (AlexJerabek) [inline types/office-js/index.d.ts]: Consider: "an object that may provide additional information if an error occurred" -> "an object with additional information about the error (if one occurred)"

File: types/office-js/index.d.ts [hunk 2]
```diff
@@ -551,6 +551,37 @@ declare namespace Office {
...
+        *  </table>
+        * 
+        * *Supported APIs*
+        * 
+        * `Office.context.mailbox.item.getCallbackTokenAsync`, `Office.context.mailbox.item.getUserIdentityTokenAsync`
+        * 
+        * *Supported errors*
+        * <table>
+        *   <tr><th>AsyncResult.error.name              </th><th>AsyncResult.error.message                                                                         </th><th>Description of diagnostics object returned                                                                                        </th></tr>
+        *   <tr><td><strong>HTTPRequestFailure</strong> </td><td>The request has failed. Please look at the diagnostics object for the HTTP error code.            </td><td>The HTTP error code in a JSON object e.g., {"HTTPCode":"401"}                                                                     </td></tr>
```
Comment (AlexJerabek) [inline types/office-js/index.d.ts]: Please put the JSON struct in code ticks.

File: types/office-js/index.d.ts [hunk 3]
```diff
@@ -551,6 +551,37 @@ declare namespace Office {
...
+        *   <tr><td><strong>InternalServerError</strong></td><td>The Exchange server returned an error. Please look at the diagnostics object for more information.</td><td>The error message from the Exchange server in a JSON object e.g., {"ErrorText": "The mailbox database is temporarily unavailable"}</td></tr>
```
Comment (AlexJerabek) [inline types/office-js/index.d.ts]: Same comment about code ticks.

File: types/office-js/index.d.ts [hunk 4]
```diff
@@ -15382,6 +15413,7 @@ declare namespace Office {
          *        asyncContext: Any state data that is passed to the asynchronous method.
          * @param callback When the method completes, the function passed in the callback parameter is called with a single parameter of 
          *                 type Office.AsyncResult. The token is provided as a string in the asyncResult.value property.
+         *                 If there was an error, then the asyncResult.error and asyncResult.diagnostics properties may provide additional information.
```
Comment (AlexJerabek) [inline types/office-js/index.d.ts]: Should asyncResult.* be in code ticks as well?

**Review by AlexJerabek — CHANGES_REQUESTED** [review]

File: types/office-js/index.d.ts (hunk 1 above)
Comment (ElizabethSamuel-MSFT) [reply types/office-js/index.d.ts]: I think that what I currently have captures the gist of it. The diagnostics object does not provide info in every error case as it's only available for those two APIs.

**Review by AlexJerabek — APPROVED** [review]

**Pull request merged**